test/Npgsql.Tests/Replication/TestDecodingReplicationTests.cs

```diff
@@ -94,14 +96,23 @@ public Task UpdateForDefaultReplicaIdentity()
                     using var streamingCts = new CancellationTokenSource();
                     var messages = rc.StartReplication(slot, streamingCts.Token).GetAsyncEnumerator();
 
+                    TestDecodingData message;
+
                     // Begin Transaction
-                    var message = await NextMessage(messages);
-                    Assert.That(message.Data, Does.StartWith("BEGIN "));
+                    while (true)
+                    {
+                        message = await NextMessage(messages);
+                        Assert.That(message.Data, Does.StartWith("BEGIN "));
 
-                    // Update
-                    message = await NextMessage(messages);
-                    Assert.That(message.Data,
-                        Is.EqualTo($"table public.{tableName}: UPDATE: id[integer]:1 name[text]:'val1'"));
+                        // Update
+                        message = await NextMessage(messages);
+                        if (message.Data.StartsWith("COMMIT "))
+                            continue;
```

Comment:
I understand that this fixes a failing test but isn't that jus covering up a problem that shouldn't happen and that we don't fully understand?

Reply:
BTW: test_decoding has the `SkipEmptyXacts` that should to the same within the logical decoding plugin.

Reply:
https://www.postgresql.org/message-id/E1XOSIv-0003rO-9d@gemulon.postgresql.org